weather_mv/loader_pipeline/pipeline.py

```diff
@@ -47,6 +47,16 @@ def pattern_to_uris(match_pattern: str, is_zarr: bool = False) -> t.Iterable[str
         yield from [x.path for x in match.metadata_list]
 
 
+def arguments_to_dict(args: t.List[str]) -> t.Dict[str, str]:
+    """Converts a list of arguments to a dictionary."""
+    result = {}
+    for i in range(0, len(args), 2):
+        key = args[i].lstrip("-")
+        value = args[i + 1]
+        result[key] = value
+    return result
```

Comment:
This will work for happy cases; won't work if someone uses command-line arguments like `--project-name="my-project"`, It's better to use some library for this. Use pipeline_options from apache_beam for this:

```
pipeline_options = PipelineOptions(args)
pipeline_options_dict = pipeline_options.get_all_options()
```

Reply:
Done.